src/mgr/FunAction.cc

```diff
@@ -49,10 +49,13 @@ Mgr::FunAction::dump(StoreEntry* entry)
 {
     debugs(16, 5, MYNAME);
     Must(entry != nullptr);
+
     if (UsingSmp())
```

Comment:
We are running in circles: We have already [removed](https://github.com/squid-cache/squid/pull/1784#discussion_r1590135674) these SMP preconditions from open/close marking code because our YAML specs do not differentiate between SMP and non-SMP cases. Please do not add these checks back, especially outside of openKidSection()/closeKidSection() implementations!

Make a choice and stick to it while fixing other problems:

* No SMP checks for any open/close markers: This option modifies existing non-SMP non-YAML reports. This PR was using this option, bugs notwithstanding, after https://github.com/squid-cache/squid/pull/1784#discussion_r1592554402 and until branch commit 414f055c99ffaf87af7ed0637aeb2832c33361aa
* No SMP checks for open/close markers in YAML reports: This option preserves existing non-SMP non-YAML reports. This option was mentioned as an alternative in https://github.com/squid-cache/squid/pull/1784#discussion_r1590135674 postscriptum but never discussed or implemented.

Your call.

Reply:
FunAction is incompatible with converting reports on it to YAML individually. This is because there is no way to register individual yaml-ness, therefore the only option available is whatever the class declares as default.

Any FunAction wanting to be converted to YAML individually will have to be converted to being its own Action, unless we extend Mgr::RegisterAction, CacheManager::registerProfile, Mgr::ActionProfile and FunActionCreator::FunActionCreator to signal yaml-ness the same way as we do for atomic-ness.

Reply:
Anyway, as of commit fbd327cc54 sections are unconditinoally opened and closed, I've validated both SMP/not SMP and atomic / not atomic.
Interestingly, since the framing is done in FunAction, all actions that use the Creator class variant of RegisterAction are on their own for framing, further diverging the two variants' behaviours.